### PR TITLE
test/e2e: add kubelet test for hostAliases with hostNetwork=true

### PR DESCRIPTION
**Pull request by neolit123**
#### What type of PR is this?

/kind feature
/area test

#### What this PR does / why we need it:

Support for this was added in 1.8 by PR #50646 but there is no e2e test.

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubernetes/issues/122420

#### Special notes for your reviewer:

the new test was added under a new Describe() with a privileged security profile on the namespace to allow for the pod to have hostnetwork=true. copies code from the agnhost test above. unclear whether that's the best setup.

works locally:

```
$ sudo ./_output/bin/e2e.test --ginkgo.focus="when hostNetwork is enabled" --kubeconfig /etc/kubernetes/admin.conf
....snip....

Will run 1 of 7418 specs
....snip....

Ran 1 of 7418 Specs in 4.596 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 7417 Skipped
PASS
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
/lgtm
/approve

The test is the same that exists for conformance but for hostNetwork true

**Review by pacoxu — COMMENTED** [review]
/lgtm

**Comment by aojea** [comment]
unhold to someone from node @pacoxu @SergeyKanzhelev ??

File: test/e2e/common/node/kubelet.go [hunk 1]
```diff
@@ -214,3 +214,48 @@ var _ = SIGDescribe("Kubelet", func() {
 		})
 	})
 })
+
+var _ = SIGDescribe("Kubelet-privileged-namespace", func() {
```
Comment (neolit123) [inline test/e2e/common/node/kubelet.go:218]: looking at other SIGDescribe() i see text without spaces:

https://github.com/neolit123/kubernetes/blob/847b2e1085f9f642292da9f3f1a56ffb5366ab86/test/e2e_node/runtimeclass_test.go#L92

`("Kubelet privileged namespace")`
should i change it like so?

Comment (neolit123) [reply test/e2e/common/node/kubelet.go]: updated https://github.com/kubernetes/kubernetes/compare/e19dde02f4f7c5ef112758e4e4d25b5c43efed32..bc83cea0eb45a051c98125466b48e99287558941

**Comment by pacoxu** [comment]
/lgtm

other sig node guys may be in vacation now😄 @bart0sh t

**Comment by aojea** [comment]
I will take the responsibility, this is more network than node  and is just copying an existing test , just using a hostNetwork pod , is not something can be controversial or anything, is a clear gap in testing coverage

/hold cancel
/lgtm
/approve

**Pull request merged**